newhelm/runners/simple_test_runner.py

```diff
@@ -145,25 +107,47 @@ class AnnotateTestItemRequest(BaseModel):
     interactions: List[PromptInteraction]
 
 
-def _collect_annotations(
-    key: str,
-    annotator: BaseAnnotator,
-    item_interactions: List[TestItemInteractions],
-    cache: BaseCache,
-) -> List[Annotation]:
-    annotations: List[Annotation] = []
-    desc = f"Collection annotations from {key}"
-    for interactions_for_item in tqdm(item_interactions, desc=desc):
-        request = AnnotateTestItemRequest(
-            interactions=interactions_for_item.interactions
-        )
+def _process_test_item(
+    item: TestItem,
+    test: BasePromptResponseTest,
+    sut: PromptResponseSUT,
+    sut_cache: BaseCache,
+    annotators: List[AnnotatorData],
+) -> TestItemRecord:
+    interactions = []
+    for prompt in item.prompts:
+        if isinstance(prompt.prompt, TextPrompt):
+            sut_request = sut.translate_text_prompt(prompt.prompt)
+        else:
+            sut_request = sut.translate_chat_prompt(prompt.prompt)
+        with sut_cache as cache:
+            sut_response = cache.get_or_call(sut_request, sut.evaluate)
+        response = sut.translate_response(sut_request, sut_response)
+        interactions.append(PromptInteraction(prompt=prompt, response=response))
+    item_interactions = TestItemInteractions(interactions=interactions, test_item=item)
```

Comment:
Do we still have a need for the `TestItemInteractions` class?

Reply:
It looks like we do not. Removed!